data_structures/heap/heap.ts

```diff
@@ -1,6 +1,88 @@
-export interface Heap<T> {
-  insert(value: T): void;
-  extract?(): T;
-  size(): number;
-  isEmpty(): boolean;
+export abstract class Heap<T> {
+  protected heap: T[];
+
+  constructor(elements: T[] = []) {
+    this.heap = [];
+    for (let element of elements) {
+      this.insert(element);
```

Comment:
Note: This is O(n log n). It is however possible to build a heap in linear time.